### PR TITLE
Add support for `textDocument/codeLens` for XML DTD

### LINKED ISSUE #252: Add support for `textDocument/codeLens` for XML DTD
Use CodeLens to display before <!ELEMENT the number of <!ATTRLIST bound to the <!ELEMENT (X references)

### PR DESCRIPTION
**Pull request by angelozerr**
Fixes #252

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
Thanks @NikolasKomonen for your review!

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/codelens/ReferenceCommand.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		super(getTitle(1), supportedByClient ? ClientCommands.SHOW_REFERENCES : "");
+		super.setArguments(Arrays.asList(uri, position));
+	}
+
+	public void increment() {
+		nbReferences++;
+		super.setTitle(getTitle(nbReferences));
+	}
+
+	private static String getTitle(int nbReferences) {
```
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/codelens/ReferenceCommand.java]: This pr was merged. Do you want i revert the commit? Or can i create a new commit?

**Comment by fbricon** [comment]
No need to revert

**Comment by angelozerr** [comment]
You mean that I change the PR and do a force push in master?

**Comment by fbricon** [comment]
No just push a new commit

**Comment by fbricon** [comment]
Never force push on master!

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/codelens/ReferenceCommand.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (angelozerr) [reply org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/codelens/ReferenceCommand.java]: Done with https://github.com/angelozerr/lsp4xml/commit/3c896455a7c7a4c979721ec3828d1845e0a2e008

**Comment by angelozerr** [comment]
> Never force push on master!

yes sure!

Done with https://github.com/angelozerr/lsp4xml/commit/3c896455a7c7a4c979721ec3828d1845e0a2e008

**Pull request merged**